test/lib/pgBackRestTest/Module/Manifest/ManifestAllTest.pm

```diff
@@ -0,0 +1,1088 @@
+        $self->testResult(sub {$oManifest->isTargetValid(BOGUS)}, false, 'isTargetValid - bError = undef, false');
+    }
+
+    ################################################################################################################################
+    if ($self->begin('dbVerion(), xactPath(), walPath()'))
+    {
+        # dbVersion, xactPath and walPath - PG < 10
+        #---------------------------------------------------------------------------------------------------------------------------
+        my $oManifest = new pgBackRest::Manifest($strBackupManifestFile, {bLoad => false, strDbVersion => PG_VERSION_94});
+
+        $self->testResult(sub {$oManifest->dbVersion()}, PG_VERSION_94, 'dbVersion < 10');
+        $self->testResult(sub {$oManifest->xactPath()}, 'pg_clog', '    xactPath - pg_clog');
+        $self->testResult(sub {$oManifest->walPath()}, 'pg_xlog', '    walPath - pg_xlog');
+
+        # dbVersion, xactPath and walPath - PG >= 10
+        #---------------------------------------------------------------------------------------------------------------------------
+        $oManifest = new pgBackRest::Manifest($strBackupManifestFile, {bLoad => false, strDbVersion => PG_VERSION_10});
+
+        $self->testResult(sub {$oManifest->dbVersion()}, PG_VERSION_10, 'dbVersion >= 10');
+        $self->testResult(sub {$oManifest->xactPath()}, 'pg_xact', '    xactPath - pg_xact');
+        $self->testResult(sub {$oManifest->walPath()}, 'pg_wal', '    walPath - pg_wal');
+    }
+
+    ################################################################################################################################
+    if ($self->begin('validate()'))
+    {
+        my $oManifest = new pgBackRest::Manifest($strBackupManifestFile, {bLoad => false, strDbVersion => PG_VERSION_94});
+
+        # Set a target:file with only a timestamp - fail size not set
+        #---------------------------------------------------------------------------------------------------------------------------
+        $oManifest->numericSet(MANIFEST_SECTION_TARGET_FILE, BOGUS, MANIFEST_SUBKEY_TIMESTAMP, 1509384645);
+        $self->testException(sub {$oManifest->validate()}, ERROR_ASSERT,
+            "manifest subvalue 'size' not set for file '" . BOGUS . "'");
+
+        # Set target:file size - fail checksum not set
+        #---------------------------------------------------------------------------------------------------------------------------
+        $oManifest->numericSet(MANIFEST_SECTION_TARGET_FILE, BOGUS, MANIFEST_SUBKEY_SIZE, 1);
+        $self->testException(sub {$oManifest->validate()}, ERROR_ASSERT,
+            "manifest subvalue 'checksum' not set for file '" . BOGUS . "'");
+
+        # Set target:file checksum - validate passes when size > 0 and checksum set
+        #---------------------------------------------------------------------------------------------------------------------------
+# CSHANG Is a zero checksum valid?
```

Comment:
No -  and it is never set to 0 currently.  This could be handled in validate, though.  Added to issue 2.